### PR TITLE
feat(password): add endpoint to verify OTP

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - we need to verify an OTP to continue the password reset process

This:
 - add an auth-server endpoint to verify the OTP

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/password.js [hunk 1]
```diff
@@ -588,6 +588,69 @@ module.exports = function (
...
+
+        request.validateMetricsContext();
+
+        const { email, code } = request.payload;
+        const account = await db.accountRecord(email);
+        const isValidCode = await otpManager.isValid(account.uid, code);
+
+        if (!isValidCode) {
+          // TODO FXA-9487
+          // increment failed attempt
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/routes/password.js:630]: The customs check doesn't increment the count for this endpoint?

**Review by bbangert — APPROVED** [review]
LGTM, one question about the customs increment.

File: packages/fxa-auth-server/lib/routes/password.js (hunk 1 above)
Comment (chenba) [reply packages/fxa-auth-server/lib/routes/password.js:630]: hmm I think the "action" can be some arbitrary string; I'll need to double check.  But I do need to add some rate limiting/blocking config for this endpoint specifically.

Either way, I was overthinking it here with this particular comment.  The logic should just be attempts in time window; there's no need to explicitly track "failed" attempts with customs.

File: packages/fxa-auth-server/lib/routes/password.ts [hunk 2]
```diff
@@ -598,6 +598,69 @@ module.exports = function (
...
+              .description(DESCRIPTION.emailRecovery),
+            code: isA
+              .string()
+              .length(config.passwordForgotOtp.digits)
+              .regex(validators.DIGITS),
+            metricsContext: METRICS_CONTEXT_SCHEMA,
+          }),
+        },
+      },
+      handler: async function (request: any) {
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/routes/password.ts:620]: Why not make it `AuthRequest` like the rest?

Comment (chenba) [reply packages/fxa-auth-server/lib/routes/password.ts:620]: The request doesn't require auth though.  And the `request.payload` type is not compatible.  (Moreover, the metrics related types, which is what I really need in this handler, are [all anys](https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/types.ts#L56-L61) anyway.

I can define a new type for that request though if that works for you.  (In addition to the metrics related types linked above, `db` is also typed as any, and so are many of the types that are accepted through the exported function.  So if we truly want to correctly type these endpoints, it'd take more than the scope of this work.)

Comment (chenba) [reply packages/fxa-auth-server/lib/routes/password.ts:620]: @bbangert haha sorry I'm not sure if your thumbs-up mean "I'm good with that explanation" or "yes, add a new type for the request" 😅

Comment (bbangert) [reply packages/fxa-auth-server/lib/routes/password.ts:620]: Doh, sorry about that. I'm good with that explanation.

**Pull request merged**